### PR TITLE
RHIDP-1925 Fix GitHub integration procedure

### PR DESCRIPTION
**Pull request by themr0c**
@rm3l said:

> I think the most important prerequisite for Bulk Import is to configure one or more GH integrations. It doesn't rely on GH authentication.
> Also, the link points to [Enabling authentication with GitHub](https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.3/html-single/authentication/index#enabling-authentication-with-github), but there is nowhere in this page how to configure a GH integration

**IMPORTANT: Do Not Merge - To be merged by Docs Team Only**

**Version(s):** 1.3.1

**Issue:** https://issues.redhat.com/browse/RHIDP-1925

**Link to docs preview:** https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-646/authentication/#enabling-authentication-with-github

**Reviews:**
- [x] SME: @rm3l 
- [x] QE: @subhashkhileri
- [x] Docs review: @jmagak 
- [ ] Additional review: @mention assignee (by writer)

**Additional information:**

### CODE REVIEW

File: modules/importing-repositories/procedure-enabling-the-bulk-import-from-github-feature.adoc [hunk 1]
```diff
@@ -3,7 +3,7 @@
 You can enable the Bulk Import feature for users and give them the necessary permissions to access it.
 
 .Prerequisites
-* You have link:{authentication-book-url}#enabling-authentication-with-github[configured GitHub authentication and integration].
+* You have link:{authentication-book-url}#enabling-authentication-with-github[configured GitHub integration].
```
Comment (rm3l) [inline modules/importing-repositories/procedure-enabling-the-bulk-import-from-github-feature.adoc:6]: Thanks for the prompt fix. There is another reference to GitHub Authentication in "Importing multiple GitHub repositories": https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-646/getting-started-rhdh/#importing-multiple-github-repositories

**Review by rm3l — APPROVED** [review]

**Comment by josephca** [comment]
@subhashkhileri please provide a review, thanks!

File: modules/authentication/proc-enabling-authentication-with-github.adoc [hunk 2]
```diff
@@ -159,6 +159,27 @@ auth:
         enterpriseInstanceUrl: ${GITHUB_HOST_DOMAIN}
 ----
 
+[TIP]
+====
+To enable GitHub integration with another authentication provider you have already configured, omit the `auth` and `signInPage` sections from this procedure:
```
[the remark this answers is not in the document]
Comment (themr0c) [reply modules/authentication/proc-enabling-authentication-with-github.adoc:164]:
```suggestion
To enable GitHub integration with another authentication provider you have already configured:

* Add the GitHub provider to the existing `auth` section.
* Keep the `signInPage` section from your authentication provider.
```

File: modules/authentication/proc-enabling-authentication-with-github.adoc [hunk 3]
```diff
@@ -159,6 +159,30 @@ auth:
...
+  github:
+    - host: ${GITHUB_HOST_DOMAIN}
+      apps:
+        - appId: ${AUTH_GITHUB_APP_ID}
+          clientId: ${AUTH_GITHUB_CLIENT_ID}
+          clientSecret: ${GITHUB_CLIENT_SECRET}
+          webhookUrl: ${GITHUB_WEBHOOK_URL}
+          webhookSecret: ${GITHUB_WEBHOOK_SECRET}
+          privateKey: |
+            ${GITHUB_PRIVATE_KEY_FILE}
```
Comment (themr0c) [inline modules/authentication/proc-enabling-authentication-with-github.adoc:182]:
```suggestion
auth:
  environment: production
  providers:
    github:
      production:
        clientId: ${AUTH_GITHUB_CLIENT_ID}
        clientSecret: ${AUTH_GITHUB_CLIENT_SECRET}
    __<your_other_authentication_providers_configuration>__
integrations:
  github:
    - host: ${GITHUB_HOST_DOMAIN}
      apps:
        - appId: ${AUTH_GITHUB_APP_ID}
          clientId: ${AUTH_GITHUB_CLIENT_ID}
          clientSecret: ${GITHUB_CLIENT_SECRET}
          webhookUrl: ${GITHUB_WEBHOOK_URL}
          webhookSecret: ${GITHUB_WEBHOOK_SECRET}
          privateKey: |
            ${GITHUB_PRIVATE_KEY_FILE}
signInPage: __<your_main_authentication_provider>__
```

File: modules/authentication/proc-enabling-authentication-with-github.adoc [hunk 4]
```diff
@@ -159,6 +159,39 @@ auth:
...
 
+[TIP]
+====
+To enable GitHub integration with another authentication provider you have already configured:
+
+* Add the GitHub provider to the existing `auth` section.
+* Keep the `signInPage` section from your authentication provider.
+
+.`app-config-rhdh.yaml` fragment with mandatory fields to enable integration with GitHub and use a different authentication provider
+[source,yaml]
```
Comment (themr0c) [inline modules/authentication/proc-enabling-authentication-with-github.adoc:170]:
```suggestion
[source,yaml,subs="+quotes"]
```

File: modules/authentication/proc-enabling-authentication-with-github.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (subhashkhileri) [reply modules/authentication/proc-enabling-authentication-with-github.adoc]: @themr0c adding the GitHub provider is not necessary, but if it's already there, there's no need to omit/remove it.

File: modules/authentication/proc-enabling-authentication-with-github.adoc [hunk 5]
```diff
@@ -159,6 +159,39 @@ auth:
         enterpriseInstanceUrl: ${GITHUB_HOST_DOMAIN}
 ----
 
+[TIP]
+====
+To enable GitHub integration with another authentication provider you have already configured:
```
Comment (hmanwani-rh) [inline modules/authentication/proc-enabling-authentication-with-github.adoc:164]:
```suggestion
To enable GitHub integration with a different authentication provider, complete the following configurations:
```

File: modules/authentication/proc-enabling-authentication-with-github.adoc [hunk 6]
```diff
@@ -159,6 +159,39 @@ auth:
...
+
+* Add the GitHub provider to the existing `auth` section.
+* Keep the `signInPage` section from your authentication provider.
```
Comment (hmanwani-rh) [inline modules/authentication/proc-enabling-authentication-with-github.adoc:167]:
```suggestion
* Keep the `signInPage` section from your authentication provider configuration.
```

File: modules/authentication/proc-enabling-authentication-with-github.adoc [hunk 7]
```diff
@@ -159,6 +159,39 @@ auth:
...
+
+.`app-config-rhdh.yaml` fragment with mandatory fields to enable integration with GitHub and use a different authentication provider
```
Comment (hmanwani-rh) [inline modules/authentication/proc-enabling-authentication-with-github.adoc:169]:
```suggestion
.`app-config-rhdh.yaml` fragment with mandatory fields to enable GitHub integration and use a different authentication provider
```

**Review by hmanwani-rh — COMMENTED** [review]
@themr0c Few comments below. Please take a look and let me know if you've any questions.

**Comment by hmanwani-rh** [comment]
@themr0c This PR needs rebasing.

**Pull request merged**